Dockerfile

```
@@ -1,112 +1,39 @@
-# If you want to change the python version, you need to change it in
-# **both** places. The python version has been left in the `environment.yml`
-# file so that people can create environments equivalent to this
-# container.
-#
-# The ordering of these steps seems to matter. You seem to have to
-# install a specific python version by hand and then pin it.
-# 1) install conda using CIVIS_MINICONDA_VERSION (may not be most up to date conda version)
-# 2) pin conda to the version given by CIVIS_CONDA_VERSION
-# 3) install the python version CIVIS_PYTHON_VERSION
-# 4) pin the python version
-#
-# Extra symlinks are added at the end because...
-#   Red Hat and Debian use different names for this file. git2R wants the latter.
-#   See conda-recipes GH 423
-RUN echo 'export PATH=/opt/conda/bin:$PATH' > /etc/profile.d/conda.sh && \
-    wget --quiet https://repo.continuum.io/miniconda/Miniconda3-${CIVIS_MINICONDA_VERSION}-Linux-x86_64.sh && \
-    /bin/bash /Miniconda3-${CIVIS_MINICONDA_VERSION}-Linux-x86_64.sh -b -p /opt/conda && \
-    rm Miniconda3-${CIVIS_MINICONDA_VERSION}-Linux-x86_64.sh && \
-    /opt/conda/bin/conda install --yes conda==${CIVIS_MINICONDA_VERSION} && \
-    conda install conda=${CIVIS_CONDA_VERSION} && \
-    echo "conda ==${CIVIS_CONDA_VERSION}" > /opt/conda/conda-meta/pinned && \
-    conda install --yes python==${CIVIS_PYTHON_VERSION} && \
-    echo "python ==${CIVIS_PYTHON_VERSION}" >> /opt/conda/conda-meta/pinned && \
-    conda clean --all -y && \
-    ln -s /opt/conda/lib/libopenblas.so /opt/conda/lib/libblas.so && \
-    ln -s /opt/conda/lib/libopenblas.so /opt/conda/lib/liblapack.so && \
-    ln -s /opt/conda/lib/libssl.so /opt/conda/lib/libssl.so.6 && \
-    ln -s /opt/conda/lib/libcrypto.so /opt/conda/lib/libcrypto.so.6
-
-# Install boto in the base environment for private s3 channel support.
-# Install Python Packages
-COPY .condarc /opt/conda/.condarc
-COPY environment.yml environment.yml
-RUN conda install -y boto && \
-    conda install -y nomkl && \
-    conda env update -f environment.yml -n root && \
-    conda clean --all -y && \
-    rm -rf ~/.cache/pip
-
-# We aren't running a GUI, so force matplotlib to use
-# the non-interactive "Agg" backend for graphics.
-# Run matplotlib once to build the font cache.
-ENV MATPLOTLIBRC=${HOME}/.config/matplotlib/matplotlibrc
-RUN mkdir -p ${HOME}/.config/matplotlib && \
-    echo "backend      : Agg" > ${HOME}/.config/matplotlib/matplotlibrc && \
-    python -c "import matplotlib.pyplot"
+COPY requirements-full.txt .
 
-# Enable widgetsnbextension for jupyter widgets.
-# See https://ipywidgets.readthedocs.io/en/stable/user_install.html.
-# This enables the extension in the conda environment. The conda-forge version
-# does this upon installation, but the default channel version doesn't seem to,
-# so we'll run this (again) just in case.
-RUN jupyter nbextension enable --py widgetsnbextension
+RUN pip install -r requirements-full.txt
```

Should we clean up like we did with the conda set-up before?

```suggestion
RUN pip install -r requirements-full.txt && \
    pip cache purge && \
    rm requirements-full.txt
```